internal/latest/tool_latest_version_finder.go

```diff
@@ -0,0 +1,186 @@
+	if strings.Contains(release.GetTagName(), toolName) {
+		return strings.ReplaceAll(release.GetTagName(), toolName+"/", "")
+	}
+	return release.GetTagName()
+}
+
+func isValidTagForTool(tag, tool string) bool {
+	if tool == version.MongoCLI {
+		return !strings.Contains(tag, version.AtlasCLI)
+	}
+	return strings.Contains(tag, tool)
+}
+
+func (s *latestReleaseVersionFinder) searchLatestVersionPerTool(currentVersion *semver.Version, toolName string) (bool, *version.ReleaseInformation, error) {
+	release, err := s.r.LatestWithCriteria(minPageSize, isValidTagForTool, toolName)
+
+	if err != nil || release == nil {
+		return false, nil, err
+	}
+
+	v, err := semver.NewVersion(versionFromTag(release, toolName))
+
+	if err != nil {
+		return false, nil, err
+	}
+
+	if currentVersion.Compare(v) < 0 {
+		return true, &version.ReleaseInformation{
+			Version:     v.Original(),
+			PublishedAt: release.GetPublishedAt().Time,
+		}, nil
+	}
+	return false, nil, nil
+}
+
+func isAtLeast24HoursPast(t time.Time) bool {
+	return !t.IsZero() && time.Since(t) >= time.Hour*24
+}
+
+func isHomebrew(tool string) bool {
+	brewFormulaPath, err := homebrewFormulaPath(tool)
+	if err != nil {
+		return false
+	}
+
+	executablePath, err := executableCurrentPath()
+	if err != nil {
+		return false
+	}
+
+	return strings.HasPrefix(executablePath, brewFormulaPath)
+}
+
+func homebrewFormulaPath(tool string) (string, error) {
+	formula := tool
+	brewFormulaPathBytes, err := exec.Command("brew", "--prefix", "--installed", formula).Output()
+	if err != nil {
+		return "", err
+	}
+
+	brewFormulaPath := strings.TrimSpace(string(brewFormulaPathBytes))
+
+	brewFormulaPath, err = filepath.EvalSymlinks(brewFormulaPath)
+	if err != nil {
+		return "", err
+	}
+
+	return brewFormulaPath, nil
+}
+
+func executableCurrentPath() (string, error) {
+	executablePath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
+	executablePath, err = filepath.EvalSymlinks(executablePath)
+	if err != nil {
+		return "", err
+	}
+
+	return executablePath, nil
+}
+
+func (s *latestReleaseVersionFinder) hasNewVersionAvailable(v, tool string) (newVersionAvailable bool, newVersion string, err error) {
+	if v == "" {
+		return false, "", nil
+	}
+	svCurrentVersion, err := semver.NewVersion(v)
+	if err != nil {
+		return false, "", err
+	}
+
+	if svCurrentVersion.Prerelease() != "" { // ignoring prerelease for code changes against master
+		*svCurrentVersion, err = svCurrentVersion.SetPrerelease("")
+		if err != nil {
+			return false, "", err
+		}
+	}
+
+	newVersionAvailable, newV, err := s.searchLatestVersionPerTool(svCurrentVersion, tool)
+	if err != nil {
+		return false, "", err
+	}
+
+	if newVersionAvailable && (!isHomebrew(tool) || isAtLeast24HoursPast(newV.PublishedAt)) {
+		return newVersionAvailable, newV.Version, nil
+	}
+
+	return false, "", nil
+}
+
+func (s *latestReleaseVersionFinder) PrintNewVersionAvailable(w io.Writer, v, tool, bin string) error {
```

Comment:
[nit] separation of concerns, `PrintNewVersionAvailable` has no relation to finding the new version , why make it a method of `latestReleaseVersionFinder`? this change makes `PrintNewVersionAvailable` have too many responsibilities, searching the latest version, considering the installation mode and then printing